packet/flatcar-linux/kubernetes/workers/workers.tf

```diff
@@ -13,8 +13,8 @@ resource "packet_device" "nodes" {
 }
 
 resource "packet_bgp_session" "bgp" {
-  count          = "${var.count}"
-  device_id      = "${element(packet_device.nodes.*.id, count.index)}"
+  count          = "${var.disable_bgp == "true" ? 0 : var.count}"
+  device_id      = "${packet_device.nodes.*.id[count.index]}"
```

Comment:
why not like this?
```suggestion
  device_id      = "${packet_device.nodes[count.index].id}"
```

Reply:
I currently have a mess of tf 0.11 and 0.12 syntax in my head, not sure if that will work. Are you sure that will have the same result?

Will try on Monday.

In any case, any reason to do it like that? Is something hidden (wouldn't be surprised if terraform has some odd behavior) there? Or what is the reason behind the suggestion? That it looks better?

Reply:
My suggestion is just a more common pattern as far as I'm aware. It saves terraform time iterating over all resources and gathering the IDs, so you only select one in the end :smile:

Reply:
Hmmm... not sure it is more common (I copied the syntax from terraform-render-bootkube repo), and for sure **I** don't know how terraform parses it, it may take as well the same time. I think you are doing many assumptions in that statement. And, in any case, it is not interesting at all to improve that.

Also, it doesn't seem to be valid syntax for terraform 0.11 what you suggest. square bracket seems to be valid on lists, and packet_device.nodes is not a list (it possible to use square brackets on terraform 0.12, though, that changes), so you need to use the splat first (as it returns a list) and then index using square bracket.

Reply:
Let me know if I'm missing something